---
title: "Extract records of interest from the EBI GWAS Catalog"
output:
  html_notebook: default
  html_document:
    df_print: paged
---
```{r setup, echo=FALSE}
library(gwascat)
library(dplyr)
library(stringr)
library(ggplot2)
library(xml2)

library(ggplot2)
library(viridis)
library(scales)

# Define project directories
projroot <- rprojroot::find_rstudio_root_file()
funcdir <- file.path(projroot, "functions")
datadir <- file.path(projroot, "data")

# Get live GWAS data
source(file.path(funcdir, "refresh_gwas_dataset.R"), chdir = TRUE)
```

# GWAS Catalog Dataset
```{r}
gwas_data %>% head()
```

## Structure
The GWAS Catalog is a large dataset (134,314x38) with records representing disease associated SNPs identified through Genome Wide Association Studies. Each table row represents a Single Nucleotide Polymorphism associated with a given disease trait. Lots of papers report multiple SNPs, so each paper tends to be replicated in a many-to-one table relationship style.

Each record contains information about the SNP of interest:

* Details of the publication in which it is reported
    + Study Title,
    + PubMed Id,    + Date Published,
    + First Author,
    + Journal me,
    + NCBI Link,
* Details of the condition with which it is associated
    + DISEASETRAIT, 
    + Risk allele, [SNP_ ID]-[Risk_Associated_Nucleotide]
    + Associated Risk (Odds Ratio or Beta coefficient, including p-value & 95% CI)
    + Sample Size, etc...
* Details of the. genetic locus at which it resides 
    + SNP ID,
    + Chromosome,
    + Genomic Region, (p/q notation),
    + Chromosomal Position, (Nucleotide index),
    + Related Genes,
    + Variant Context, (SNP Functional class - exon/missense/non-coding/etc)
    + Platform (Genotyping platformI manufacturer used in Stage 1)
    
### Reducing the dataset
First we need to reduce the full GWAS Catalog [data dictionary here](https://www.ebi.ac.uk/gwas/d.  ocs/fileheaders) down to include only cancer related records.

Once this is done, each record will be assessed and categorised based on a thematic analysis of the study title.

The intent here is to isolate only those records that have some kind of relevance to cancer. This will be done by filtering for cancer-like keywords in the DISEASE.TRAIT field.

#### Positive RegEx matches (include)
* "cancer"
    + Match the pattern "cancer"
* "oma([^\\w]|$)"
    + Match the pattern "oma" only where it is not followed by a "word character" (i.e. ASCII letter, digit or underscore) OR where it is followed by a line ending. Effectively matches any word with the suffix "oma"
    + Included: Melan**oma**, Carcin**oma**, etc...
    + Not included: Chr**oma**tosis, Bi**oma**rker, S**oma**tic 
    + **NOTE:** *Need to validate that no cancer records are excluded*

#### Negative RegEx matches (exclude)
* "glaucoma"
    + Exclude the pattern "glaucoma"
    + Glaucoma is a degenerative eye syndrome characterised by vision loss as a result of damage to the optic nerve (not cancer).
* "tripanosoma"
    + Exclude the pattern "tripanosoma"
    + Tripanosoma is a genus of parasitic protozoa (not cancer).
* "hematoma"
    + Exclude the pattern "hematoma"
    + A hematoma is an abnormal collection of blood  external to the circulatory system (a more serious form of bruise) caused by the leakage of blood from a *large* blood vessel (not cancer).

```{r, message=FALSE}
source(file.path(funcdir, "select_gwas_cancer.R"))
extract_cancer_data(gwas_data)
```

```{r}
# Prepare titles for classification
exclude <- c(
    "body mass index",
    "cholangitis",
    "insulinoma",
    "pancreatitis",
    "smoking",
    "smoker",
    "interaction",
    "response",
    "therapy",
    "treatment",
    "radiotherapy",
    "radiation",
    "chemotherapy",
    "toxicity",
    "pain",
    "platinum",
    "anastrozole",
    "asparaginase",
    "bevacizumab",
    "carboplatin",
    "cetuximab",
    "cisplatin",
    "docetaxel",
    "gemcitabine",
    "irinotecan",
    "paclitaxel",
    "taxane",
    "trastuzumab",
    "alopecia",
    "tamoxifen",
    "hormone therapy",
    "estradiol",
    "estrone",
    "androstene",
    "thiopurine",
    "asparaginase",
    "vincristine",
    "methotrexate",
    "kinetics",
    "childhood", # excludes SNPs predicting ALL in childhood, no other conflicts
    "free",
    "recurrence",
    "survival",
    "metastasis",
    "aggressiveness",
    "mortality",
    "infection",
    "hepatitis",
    "benign",
    "adenoma",
    "neuropathy"
) %>% 
    paste(collapse="|")

# cancer_classes <- 
gwas_cancer_data %>% 
	dplyr::select(PUBMEDID, DISEASE.TRAIT) %>%
	# dplyr::select(PUBMEDID, DISEASE.TRAIT, MAPPED_TRAIT)# %>%
	mutate(
	    traits =             tolower(DISEASE.TRAIT)# all lowercase,
	    ) %>% 
    filter(!str_detect(traits, exclude)) %>%
    mutate(
	    biliary =            if_else(str_detect(traits, "cholangio|gall"), TRUE, FALSE),     # exclude cholangiocarcinomas reported as a side effect of PSC
	    bladder =            if_else(str_detect(traits, "(^| )bladder"), TRUE, FALSE),
	    cns =                if_else(str_detect(traits, "glio|mening"), TRUE, FALSE),
	    breast =             if_else(str_detect(traits, "breast"), TRUE, FALSE),
	    cervical =           if_else(str_detect(traits, "cervical"), TRUE, FALSE),
	    colorectal =         if_else(str_detect(traits, "colo"), TRUE, FALSE),
	    endometrial =        if_else(str_detect(traits, "endometri"), TRUE, FALSE),
	    oesophageal =        if_else(str_detect(traits, "esoph"), TRUE, FALSE),
	    gastric =            if_else(str_detect(traits, "gastric"), TRUE, FALSE),
	    head_neck =          if_else(str_detect(traits, "oral cavity|aryn"), TRUE, FALSE),
	    kidney =             if_else(str_detect(traits, "renal"), TRUE, FALSE),
	    leukemia =           if_else(str_detect(traits, "leukemia"), TRUE, FALSE),
	    liver =              if_else(str_detect(traits, "hepato"), TRUE, FALSE),
	    lung =               if_else(str_detect(traits, "lung"), TRUE, FALSE),
	    lymphoma =           if_else(str_detect(traits, "lymphom"), TRUE, FALSE),
	    melanoma =           if_else(str_detect(traits, "(^|^-|\\s)melanoma"), TRUE, FALSE),
	    multiple_myeloma =   if_else(str_detect(traits, "myeloma"), TRUE, FALSE),
	    neuroblastoma =      if_else(str_detect(traits, "neuroblastoma"), TRUE, FALSE),
	    ovarian =            if_else(str_detect(traits, "ovarian"), TRUE, FALSE),
	    pancreatic =         if_else(str_detect(traits, "pancrea"), TRUE, FALSE),
	    prostate =           if_else(str_detect(traits, "prostate"), TRUE, FALSE),
	    sarcoma =            if_else(str_detect(traits, "sarcoma"), TRUE, FALSE),
	    skin_non_melanoma =  if_else(str_detect(traits, "skin|basal|keratinocyte|squamous"), TRUE, FALSE),
	    thyroid =            if_else(str_detect(traits, "thyroid"), TRUE, FALSE),
	    testicular =         if_else(str_detect(traits, "testic"), TRUE, FALSE),
	    cancer_multiple_or_unspecified = if_else(str_detect(traits, "^cancer"), TRUE, FALSE),
	) %>% 
    mutate(
	    total              = rowSums(.[4:29]),
	    cancer_multiple_or_unspecified = if_else(total>1, TRUE, cancer_multiple_or_unspecified)
    ) %>% 
    mutate( # Exclude anything marked as being associated with multiple or unspecified cancer type as not being associated with a specific type
        biliary =            if_else(cancer_multiple_or_unspecified == TRUE, FALSE, biliary),
	    bladder =            if_else(cancer_multiple_or_unspecified == TRUE, FALSE, bladder),
	    cns =                if_else(cancer_multiple_or_unspecified == TRUE, FALSE, cns),
	    breast =             if_else(cancer_multiple_or_unspecified == TRUE, FALSE, breast),
	    cervical =           if_else(cancer_multiple_or_unspecified == TRUE, FALSE, cervical),
	    colorectal =         if_else(cancer_multiple_or_unspecified == TRUE, FALSE, colorectal),
	    endometrial =        if_else(cancer_multiple_or_unspecified == TRUE, FALSE, endometrial),
	    oesophageal =        if_else(cancer_multiple_or_unspecified == TRUE, FALSE, oesophageal),
	    gastric =            if_else(cancer_multiple_or_unspecified == TRUE, FALSE, gastric),
	    head_neck =          if_else(cancer_multiple_or_unspecified == TRUE, FALSE, head_neck),
	    kidney =             if_else(cancer_multiple_or_unspecified == TRUE, FALSE, kidney),
	    leukemia =           if_else(cancer_multiple_or_unspecified == TRUE, FALSE, leukemia),
	    liver =              if_else(cancer_multiple_or_unspecified == TRUE, FALSE, liver),
	    lung =               if_else(cancer_multiple_or_unspecified == TRUE, FALSE, lung),
	    lymphoma =           if_else(cancer_multiple_or_unspecified == TRUE, FALSE, lymphoma),
	    melanoma =           if_else(cancer_multiple_or_unspecified == TRUE, FALSE, melanoma),
	    multiple_myeloma =   if_else(cancer_multiple_or_unspecified == TRUE, FALSE, multiple_myeloma),
	    neuroblastoma =      if_else(cancer_multiple_or_unspecified == TRUE, FALSE, neuroblastoma),
	    ovarian =            if_else(cancer_multiple_or_unspecified == TRUE, FALSE, ovarian),
	    pancreatic =         if_else(cancer_multiple_or_unspecified == TRUE, FALSE, pancreatic),
	    prostate =           if_else(cancer_multiple_or_unspecified == TRUE, FALSE, prostate),
	    sarcoma =            if_else(cancer_multiple_or_unspecified == TRUE, FALSE, sarcoma),
	    skin_non_melanoma =  if_else(cancer_multiple_or_unspecified == TRUE, FALSE, skin_non_melanoma),
	    thyroid =            if_else(cancer_multiple_or_unspecified == TRUE, FALSE, thyroid),
	    testicular =         if_else(cancer_multiple_or_unspecified == TRUE, FALSE, testicular),
    )
```

```{r}
gwas_cancer_data %>% 
	# dplyr::select(PUBMEDID, DISEASE.TRAIT) %>% 
	dplyr::select(
	    # PUBMEDID, 
	    DISEASE.TRAIT, 
	    MAPPED_TRAIT
	) %>%
    # count(
    #     # PUBMEDID, 
    #     MAPPED_TRAIT
    # ) %>% arrange(MAPPED_TRAIT) %>% 
    filter(
    # count(
        !str_detect(MAPPED_TRAIT, "response to"),
        !str_detect(MAPPED_TRAIT, "mortality"),
        !str_detect(MAPPED_TRAIT, "metastasis"),
        !str_detect(MAPPED_TRAIT, "survival"),
        !str_detect(MAPPED_TRAIT, "benign"),
        # str_detect(MAPPED_TRAIT, "")
    )
```
```{r}
gwas_cancer_data %>% filter(str_detect(MAPPED_TRAIT, "smoking status measurement"))
```


```{r}
# Check whether any publications are not classified
cancer_classes %>% 
    filter(
        biliary == FALSE,
        bladder == FALSE,
        cns == FALSE,
        breast == FALSE,
        cervical == FALSE,
        colorectal == FALSE,
        endometrial == FALSE,
        oesophageal == FALSE,
        gastric == FALSE,
        head_neck == FALSE,
        kidney == FALSE,
        leukemia == FALSE,
        liver == FALSE,
        lung == FALSE,
        lymphoma == FALSE,
        melanoma == FALSE,
        multiple_myeloma == FALSE,
        neuroblastoma == FALSE,
        ovarian == FALSE,
        pancreatic == FALSE,
        prostate == FALSE,
        sarcoma == FALSE,
        skin_non_melanoma == FALSE,
        thyroid == FALSE,
        testicular == FALSE,
        cancer_multiple_or_unspecified == FALSE,
    ) %>% distinct(PUBMEDID, DISEASE.TRAIT)

```

```{r empty df to store identified SNP records}
selected <- matrix(ncol=length(colnames(cancer_classes)),nrow=0) %>% 
    as.data.frame() %>%
    setNames(colnames(cancer_classes))
```

```{r select relevant biliary tract (bile duct/gallbladder) SNPs}
selected <- cancer_classes %>% 
	filter(
	    biliary == TRUE,
	    # !str_detect(traits, "cholangitis"),
	) %>% 
    rbind(selected)
```

```{r select relevant bladder SNPs}
selected <- cancer_classes %>% 
	filter(
	    bladder == TRUE,
 		# !str_detect(traits, "smoking") # exclude publications on smoking interaction - one pub: 24662972
 	) %>% 
    rbind(selected)
```

```{r select relevant breast SNPs}
# chemotherapy <- "anastrozole|taxane|paclitaxel|trastuzumab|alopecia|bevacizumab"
# 	# anastrozole - one publication:  30648747 (phamacokinetics/pharmacogenomics - SNP encodes anastrozole influx transporter)
# 	# taxane      - one publication:  29278617
# 	# paclitaxel  - two publications: 28763429, 24025145 (adverse effects: alopecia/change in LVEF)
# 	# trastuzumab - one publication:  28763429 (adverse effects: change in LVEF)

# hormone_therapy <- "tamoxifen|hormone therapy"
# 	# tamoxifen   - two publications: 30161160, 22180457
# 
# radiation <- "childhood"
# 	# childhood   - one publication:  29059430 (studied effect of chest-directed radiotherapy on female survivors of childhood cancer
# 
# survival_time <- "free"
# 	# free        - two publications: 27758888, 25867717 (Cancer-free interval post treatment)
# 
# tumour_biochem <- "estradiol|estrone|androstene"
#     # estradiol 23518928
# 	# estrone     - one publication:  28429243 (hormone levels in resected early-stage tumours)
# 	# androstene  - one publication:  28429243

selected <- cancer_classes %>%
    filter(
        breast == TRUE,
        # !str_detect(traits, "mortality"),          # exclude publications targeting post-diagnosis event (mortality)     - one publication:   30787463
        # !str_detect(traits, chemotherapy),         # exclude publications targeting treatment response (chemotherapies)  - four publications: see exclusion terms
        # !str_detect(traits, hormone_therapy),      # exclude publications targeting treatment response (hormone therapy) - two publications:  see exclusion terms
        # !str_detect(traits, radiation),            # exclude publications targeting treatment exposure (radiation)       - one publication :  see exclusion terms
        # !str_detect(traits, survival_time),        # exclude publications characterising recurrent disease               - two publications:  see exclusion terms
        # !str_detect(traits, tumour_biochem),       # exclude publications characterising tumour environment              - one publication :  see exclusion terms
        # !str_detect(traits, "survival|prognosis"), # exclude publications predicting post-diagnosis survival             - six publications:  29423119, 25964295, 25890600, 25867717, 25526632, 22232737, 23319801
    ) %>% 
    rbind(selected)

# rm(
#     "chemotherapy",
#     "hormone_therapy",
#     "radiation",
#     "survival_time",
#     "tumour_biochem"
#    )
```

```{r select relevant cervical SNPs}
selected <- cancer_classes %>% 
	filter(
	    cervical == TRUE,
	    # !str_detect(traits, "chemotherapy"),        # exclude publications targeting treatment response - one publication: 28120872
	) %>% 
    rbind(selected)
```

```{r select relevant cns (brain)) SNPs}
selected <- cancer_classes %>%
	filter(
	    cns == TRUE,
 	 ) %>%
    rbind(selected)
```

```{r select relevant colorectal SNPs}
# survival <- "metastasis|survival"
# 	# metastasis - one publication:    30738427
# 	# survival   - three publications: 26586795, 26222057, 25866641

selected <- cancer_classes %>%
    filter(
        colorectal == TRUE,
        # !str_detect(traits, survival),            # exclude publications targeting time to non-diagnosis event        - three publications: see exclusion terms
        # !str_detect(traits, "toxicity"),          # exclude publications targeting adverse treatment response         - one publications:   30557370
        # !str_detect(traits, "adenoma"),           # exclude publications targeting benign traits                      - one publication:    29228715
        # !str_detect(traits, "cholangitis"),       # exclude publications targeting disease interaction                - one publication:    28779025
        # !str_detect(traits, "interaction"),     # exclude publications targeting exposure factors                   - one publication:    27723779, 26766742, 25781442, 24743840
    ) %>% 
    rbind(selected)

# rm("survival")
```

```{r select relevant endometrial SNPs}
selected <- cancer_classes %>% 
	filter(
	    endometrial == TRUE,
	) %>% 
    rbind(selected)
```

```{r select relevant gastric SNPs}
# selected <- 
cancer_classes %>%
	filter(
	    gastric == TRUE,
 	) #%>% 
    rbind(selected)
```

```{r select relevant head/neck SNPs}
selected <- cancer_classes %>%
	filter(
	    head_neck == TRUE,
	    # !str_detect(traits, "radiation") # exclude adverse treatment effects - 1 pub 30299488
 	) %>% 
    rbind(selected)
```

```{r select relevant kidney SNPs}
selected <- cancer_classes %>%
	filter(
	    kidney == TRUE,
 	) %>% 
    rbind(selected)
```

```{r select relevant leukemia (blood) SNPs}
selected <- cancer_classes %>%
	filter(
	    leukemia == TRUE,
	    # !str_detect(traits, "response|therapy|thiopurine|asparaginase|vincristine|methotrexate|kinetics")
	) %>%
    rbind(selected)
```

```{r select relevant liver SNPs}
selected <- cancer_classes %>%
	filter(
	    liver == TRUE,
	    # !str_detect(traits, "response|survival|hepatitis")
	) %>%
    rbind(selected)
```

```{r select relevant lung SNPs}
selected <- cancer_classes %>%
	filter(
	    lung == TRUE,
	    # !str_detect(traits, "smoker|interaction|platinum|irinotecan|survival|recurrence")
 	) %>% 
    rbind(selected)
```

```{r select relevant lymphoma SNPs}
selected <- cancer_classes %>%
	filter(
	    lymphoma == TRUE,
	    # !str_detect(traits, "survival")
	) %>% 
    rbind(selected)
```

```{r select relevant multiple myeloma (blood) SNPs}
selected <- cancer_classes %>%
	filter(
	    multiple_myeloma == TRUE,
	    # !str_detect(traits, "survival|infection|neuropathy"),        # exclude publications targeting myeloma-secondary event - three publications: 28120872, 29594489, 28317148
	) %>% 
    rbind(selected)
```

```{r select relevant melanoma SNPs}
selected <- cancer_classes %>%
	filter(
	    melanoma == TRUE,
	    # !str_detect(traits, "survival|infection|neuropathy"),        # exclude publications targeting myeloma-secondary event - three publications: 28120872, 29594489, 28317148
	) %>% 
    rbind(selected)
```

```{r select relevant neuroblastoma SNPs}
selected <- cancer_classes %>%
	filter(
	    neuroblastoma == TRUE,
 	) %>%
    rbind(selected)
```

```{r select relevant oesophageal SNPs}
selected <- cancer_classes %>% 
	filter(
	    oesophageal == TRUE,
	    # !str_detect(traits, "interaction|survival")
 	) %>% 
    rbind(selected)
```

```{r select relevant ovarian SNPs}
# chemotherapy <- "carboplatin|paclitaxel"
# 	# carboplatin - three publications 29367611 (pharmacokinetics), 28935272 (progression-free survival), 28935272 (treatment response)
# 	# paclitaxel  - three publications 29367611 (pharmacokinetics), 28935272 (progression-free survival), 28935272 (treatment response)

selected <- cancer_classes %>%
	filter(
	    ovarian == TRUE,
	    # !str_detect(traits, chemotherapy),        # exclude publications targeting treatment response         - one publication:  see exclusion list
	) %>% 
    rbind(selected)

# rm("chemotherapy")
```

```{r select relevant pancreatic SNPs}
selected <- cancer_classes %>%
	filter(
	    pancreatic == TRUE,
	    # !str_detect(traits, "gemcitabine|survival|pancreatitis")
 	) %>% 
    rbind(selected)
```


```{r select relevant prostate SNPs}
# prognosis <- "aggressiveness|survival"
# 	# aggressiveness - one publication: 25939597
# 	# survival       - one publication: 26307654
# 
# treatment <- "treatment|docetaxel|radiotherapy"
# 	# treatment  - one publication: 20932654
#  	# docetaxel  - one publication: 27143689
#     # radiotherapy - three publications: 27515689, 24974847, 23376709

 selected <- cancer_classes %>% 
	filter(
	    prostate == TRUE,
	    # !str_detect(traits, 'benign'),          # exclude publications targeting benign traits             - one publication:    30410027
	    # !str_detect(traits, treatment),         # exclude publications targeting adverse treatment effects - five publications:    see exclusion terms
	    # !str_detect(traits, prognosis),         # exclude publications targeting prognosis                 - two publications:   see exclusion terms
	) %>%
    rbind(selected)

# rm("prognosis",
#    "treatment")
```

```{r select relevant sarcoma SNPs}
selected <- cancer_classes %>%
	filter(
	    sarcoma == TRUE,
	    # !str_detect(traits, "survival|metastasis")
 	) %>% 
    rbind(selected)
```

```{r select relevant skin SNPs}
selected <- cancer_classes %>%
	filter(
	    skin_non_melanoma == TRUE,
	    # !str_detect(traits, "toxicity|pain|survival")
 	) %>% 
    rbind(selected)
```

```{r select relevant thyroid SNPs}
selected <- cancer_classes %>%
	filter(
	    thyroid == TRUE,
 		# !str_detect(traits, "radiation"), # exclude publications with additional exposure factors - one publication: 20350937
 	) %>% 
    rbind(selected)
```

```{r select relevant testicular SNPs}
selected <- cancer_classes %>%
	filter(
	    testicular == TRUE,
 	) %>% 
    rbind(selected)
```

```{r select relevant "Cancer" SNPs}
selected <- cancer_classes %>%
	filter(
	    cancer_multiple_or_unspecified == TRUE,
 	) %>% 
    rbind(selected)
```

```{r bind}
gwas_classified <- selected %>%
	inner_join(gwas_cancer_data, by = c("PUBMEDID","DISEASE.TRAIT")) %>%
	mutate(
	    biliary = if_else(is.na(biliary), FALSE, biliary),
	    bladder = if_else(is.na(bladder), FALSE, bladder),
	    cns = if_else(is.na(cns), FALSE, cns),
	    breast = if_else(is.na(breast), FALSE, breast),
	    cervical = if_else(is.na(cervical), FALSE, cervical),
	    colorectal = if_else(is.na(colorectal), FALSE, colorectal),
	    endometrial = if_else(is.na(endometrial), FALSE, endometrial),
	    oesophageal = if_else(is.na(oesophageal), FALSE, oesophageal),
	    gastric = if_else(is.na(gastric), FALSE, gastric),
	    head_neck = if_else(is.na(head_neck), FALSE, head_neck),
	    kidney = if_else(is.na(kidney), FALSE, kidney),
	    leukemia = if_else(is.na(leukemia), FALSE, leukemia),
	    liver = if_else(is.na(liver), FALSE, liver),
	    lung = if_else(is.na(lung), FALSE, lung),
	    lymphoma = if_else(is.na(lymphoma), FALSE, lymphoma),
	    melanoma = if_else(is.na(melanoma), FALSE, melanoma),
	    multiple_myeloma = if_else(is.na(multiple_myeloma), FALSE, multiple_myeloma),
	    neuroblastoma = if_else(is.na(neuroblastoma), FALSE, neuroblastoma),
	    ovarian = if_else(is.na(ovarian), FALSE, ovarian),
	    pancreatic = if_else(is.na(pancreatic), FALSE, pancreatic),
	    prostate = if_else(is.na(prostate), FALSE, prostate),
	    sarcoma = if_else(is.na(sarcoma), FALSE, sarcoma),
	    skin_non_melanoma = if_else(is.na(skin_non_melanoma), FALSE, skin_non_melanoma),
	    thyroid = if_else(is.na(thyroid), FALSE, thyroid),
	    testicular = if_else(is.na(testicular), FALSE, testicular),
	    cancer_multiple_or_unspecified = if_else(is.na(cancer_multiple_or_unspecified), FALSE, cancer_multiple_or_unspecified),
	) %>%
    distinct(PUBMEDID, SNPS,
             biliary,
             bladder,
             breast,
             cervical,
             cns,
             colorectal,
             endometrial,
             gastric,
             head_neck,
             kidney,
             leukemia,
             liver,
             lung,
             lymphoma,
             melanoma,
             multiple_myeloma,
             neuroblastoma,
             oesophageal,
             ovarian,
             pancreatic,
             prostate,
             sarcoma,
             skin_non_melanoma,
             thyroid,
             testicular,
             cancer_multiple_or_unspecified,
             .keep_all = TRUE) %>% 
	arrange(PUBMEDID)
```

```{r}
gwas_classified <- gwas_classified %>% 
    tidyr::gather(cancer_class, tf, biliary:cancer_multiple_or_unspecified) %>% 
    filter(tf==1) %>% 
    dplyr::select(-c(tf, traits, start:strand)) %>% 
    arrange(cancer_class) %>% 
    write_csv(file.path(projroot, "data/gwas_cancer_classifications.csv"))
```

```{r}
plot_data <- gwas_classified %>% 
    distinct(SNPS, cancer_class) %>% 
    count(cancer_class) %>% 
    mutate(
        cancer_class = case_when(
            cancer_class == "cancer_multiple_or_unspecified" ~ "Multiple Cancers",
            cancer_class == "head_neck" ~ "Head/Neck",
            cancer_class == "skin_non_melanoma" ~ "Skin",
            cancer_class == "multiple_myeloma" ~ "Multiple Myeloma",
            TRUE ~ cancer_class
        ),
        `Cancer Class` = str_to_title(cancer_class),
        `Cancer Class` = str_replace(`Cancer Class`, "Cns", "CNS"),
    ) %>% 
    dplyr::rename("Distinct SNPs" = n) %>%
    mutate(
        strat = case_when(
            `Distinct SNPs` > 250              ~ "250+",
            between(`Distinct SNPs`, 101, 250) ~ "101-250",
            between(`Distinct SNPs`,  51, 100) ~ "51-100",
            `Distinct SNPs` <= 50               ~ "0-50",
            ) %>% factor(levels = c("0-50", "51-100", "101-250", "250+"))
    )
```

```{r}
plot_data %>% 
    count(strat)
```
```{r, fig.width=20, fig.height=12}
plot_data %>% 
    filter(strat == "0-50") %>% 
    ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
    theme_minimal() +
    geom_bar(stat = "identity") + 
    geom_text(aes(label = `Cancer Class`), vjust=1.6, color="white", size=10) + 
        theme(
            axis.text.x  = element_blank(),
            axis.title.x = element_blank(),
            axis.title.y = element_text(angle = 45, vjust=0.5),
            legend.position = "none"
            ) +
        scale_x_discrete(name = "", breaks = NULL) +
        scale_y_continuous(limits = c(0,50), breaks = NULL,oob = rescale_none , name = "") +
        scale_fill_viridis(discrete = T, option = "plasma")

ggsave("images/lt_50_overlay.png", width = 20, height = 4, bg = "transparent")
```

```{r, fig.width=20, fig.height=8}
plot_data %>% 
    filter(strat == "51-100") %>% 
    ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
    theme_minimal() +
    geom_bar(stat = "identity") + 
    geom_text(aes(label = `Cancer Class`), vjust=1.6, color="white", size=10) + 
        theme(
            axis.text.x  = element_blank(),
            axis.title.x = element_blank(),
            axis.title.y = element_text(),
            rect = element_rect(fill = "transparent"),
            legend.position = "none"
            ) +
        scale_x_discrete(name = "", breaks = NULL) +
        scale_y_continuous(limits = c(0,100), breaks = NULL, oob = rescale_none , name = "") +
        scale_fill_viridis(discrete = T, direction = -1)

ggsave("images/lt_100_midlay.png", width = 20, height = 8, bg = "transparent")
```

```{r, fig.width=20, fig.height=12}
plot_data %>%
    filter(strat == "101-250") %>%
    ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
    theme_minimal() +
    geom_bar(stat = "identity") +
    geom_text(aes(label = `Cancer Class`), vjust=1.6, color="white", size=10) +

        theme(
            axis.text.x = element_blank(),
            axis.title.y = element_text(angle = 45, vjust = 0.5),
            rect = element_rect(fill = "transparent"),
            legend.position = "none"
            ) +
        scale_x_discrete(name = "", breaks = NULL) +
        scale_y_continuous(limits = c(0, 250), breaks = NULL, oob = rescale_none, name = "") +
        scale_fill_viridis(discrete = T)

ggsave("images/lt_250_underlay.png", width = 20, height = 12, bg="transparent")
```

```{r, fig.width=20, fig.height=12}
plot_data %>% 
    dplyr::filter(strat == "250+") %>% arrange(desc(cancer_class)) %>% 
    ggplot(mapping = aes(x = dplyr::desc(`Cancer Class`), y=`Distinct SNPs`, fill = `Cancer Class`)) +
    # ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
    theme_minimal(base_size=22) +
    geom_bar(stat = "identity") + 
    geom_text(aes(label = `Cancer Class`), vjust=1.6, color="white", size=10) + 
        theme(
            axis.text.x  = element_blank(), 
            axis.title.x = element_blank(), 
            # axis.text = element_text(size = 10),
            # axis.ticks.x = element_blank(),
            # axis.title.y = element_text(),
            rect = element_rect(fill = "transparent"),
            legend.position = "none",
            legend.text = element_blank()
            ) +
        scale_x_discrete(name = "", breaks=NULL) +
        scale_y_continuous(name = "Distinct SNPs", limits = c(40,875), breaks = c(seq(0,45,5), seq(50,90,10),seq(100,900,50)), oob = rescale_none, position = "right") +
        scale_fill_viridis(discrete=T, option = "plasma") #+
    # ylab("Distinct SNPs")

ggsave("images/gt_250_sidelay.png", width = 20, height = 43.2, bg = "transparent")
```



```{r, fig.width=20, fig.height=12}
# plot_data %>% 
#     filter(strat == "0-50") %>% 
#     ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
#     theme_minimal() +
#         geom_bar(stat = "identity") + 
#         theme(
#             # axis.text.x = element_text(angle = -30),
#             axis.title.y = element_text(angle = 45, vjust=0.5),
#             legend.position = "none"
#             ) +
#         scale_y_continuous(limits = c(2,50), breaks = seq(0,900,5),oob = rescale_none , name = "") +
#         scale_fill_viridis(discrete = T)
# 
# ggsave("images/lt_50.png", width = 20, height = 4)
```

```{r, fig.width=20, fig.height=12}
# lt_100 <- plot_data %>% 
#     filter(strat == "51-100") %>% 
#     ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
#     theme_minimal() +
#         geom_bar(stat = "identity") + 
#         theme(
#             #axis.text.x  = element_text(angle = -30),
#             axis.title.y = element_text(),
#             axis.title.x = element_blank(),
#             rect = element_rect(fill = "transparent"),
#             legend.position = "none"
#             ) +
#         scale_x_discrete(name = "") +
#         scale_fill_viridis(option="plasma", discrete = T)
# 
# lt_100 + 
#     scale_y_continuous(limits = c(50,100), breaks = seq(0,900,10), oob = rescale_none)
# ggsave("images/lt_100.png", width = 20, height = 4, bg = "transparent")
# 
# 
# lt_100 +
#     scale_y_continuous(limits = c(0,100), breaks = seq(0,100,10), oob = rescale_none)
# ggsave("images/lt_100_underlay.png", width = 20, height = 8, bg = "transparent")
```

```{r, fig.width=20, fig.height=12}
# lt_250 <- plot_data %>% 
#     # filter(strat %in% c("0-50", "51-100", "101-250")) %>% 
#     filter(strat == "101-250") %>%
#     ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
#     theme_minimal() +
#         geom_bar(stat = "identity") + 
#         theme(
#             # axis.text.x = element_text(angle = -30),
#             axis.title.y = element_text(angle = 45, vjust = 0.5),
#             rect = element_rect(fill = "transparent"),
#             legend.position = "none"
#             ) +
#         scale_x_discrete(name = "") +
#         scale_fill_viridis(discrete = T)
# 
# lt_250 +        
#     scale_y_continuous(limits = c(100, 250), breaks = seq(0,900,50), oob = rescale_none, name = "")
# ggsave("images/lt_250.png", width = 20, height = 4, bg = "transparent")
# 
# lt_250 +
#     scale_y_continuous(limits = c(10, 250), breaks = seq(0,900,50), oob = rescale_none, name = "")
# ggsave("images/lt_250_underlay.png", width = 20, height = 12, bg = "transparent")
```

```{r, fig.width=10, fig.height=12}
# gt_250 <- plot_data %>% 
#     filter(strat == "250+") %>% 
#     ggplot(mapping = aes(x = `Cancer Class`, y=`Distinct SNPs`, fill = `Cancer Class`)) +
#     theme_minimal() +
#     geom_bar(stat = "identity") + 
#     geom_text(aes(label = `Cancer Class`), vjust=1.6, color="white", size=5) + 
#         theme(
#             axis.text.x = element_blank(), 
#             axis.title.y = element_blank(),
#             rect = element_rect(fill = "transparent"),
#             legend.position = "none",
#             legend.text = element_blank()
#             ) +
#         scale_fill_viridis(option="plasma", discrete=T)
# 
# gt_250 +
#     scale_y_continuous(position = "right", limits = c(250,875), breaks = seq(0,900,50), oob = rescale_none) +
# ggsave("images/gt_250.png", width = 10, height = 12, bg = "transparent")
# 
# gt_250 +
#     scale_y_continuous(position = "right", limits = c(40,875), breaks = seq(0,900,50), oob = rescale_none) +
# ggsave("images/gt_250_underlay.png", width = 10, height = 12, bg = "transparent")
```
